### ISSUE TITLE
Verify user account has password or ssh key

### ISSUE THREAD

**Issue by mdhorn** [clearlinux/clr-installer#221]
**Describe the bug**
clr-installer allow putting a used account in the configuration file without a password or ssh key.

**To Reproduce**
1. Make a copy of a simple image config file like scripts/live.yaml
2. Add a user with just the login and username
```
users:
    - login: joe
      username: Joe User
```

3. Generate an image with the configuration file
`clr-installer -l 4 -c scripts/live-with-user.yaml`
4. Image is generated.

**Expected behavior**
The installer should fail with a message requiring either a password or an SSH key for the user.

**Environment (please complete the following information):**
 - Clear Linux OS Version: [swupd info]
```
Installed version: 27800
Version URL:       https://cdn.download.clearlinux.org/update/
Content URL:       https://cdn.download.clearlinux.org/update/
```
 - Installer Version [clr-installer -v]
`clr-installer: 1.1.4`
 - Platform: [hardware infomation]
Any

**Comment by karthikprabhuvinod** [comment]
@mdhorn Isn't this something that needs to be done when we perform UnMarshal the yaml and if it does not contain password, we error out??

**Comment by mdhorn** [comment]
> @mdhorn Isn't this something that needs to be done when we perform UnMarshal the yaml and if it does not contain password, we error out??

No, we do not want to check during the unmarshaling as we always parse a YAML configuration file for CLI, GUI, and TUI. For the GUI/TUI, we get our defaults from the YAML configuration file.
We would want to better validate the user-related fields in the CLI (massinstall) mode after parsing before starting the install.
